src/sugar3/activity/activity.py

```diff
@@ -436,13 +436,23 @@ def __init__(self, handle, create_jobject=True):
                                            self.__jobject_updated_cb)
         self.set_title(self._jobject.metadata['title'])
 
+        if 'SUGAR_VERSION' not in os.environ:
+            bundle = get_bundle_instance(get_bundle_path())
+            self.set_icon_from_file(bundle.get_icon())
+
+
     def run_main_loop(self):
         Gtk.main()
 
     def _initialize_journal_object(self):
         title = _('%s Activity') % get_bundle_name()
-        settings = Gio.Settings('org.sugarlabs.user')
-        icon_color = settings.get_string('color')
+       
+        if 'org.sugarlabs.user' in Gio.Settings.list_schemas():
```

Comment:
Couldn't the launcher just set up the gsettings properly?  This feels very dirty - many things expect the gsettings to work.